### PR TITLE
Refactor/textinput adornment

### PR DESCRIPTION
**Pull request by matkoson**

### CODE REVIEW

File: src/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -229,12 +221,19 @@ class TextInput extends React.Component<TextInputProps, State> {
   ref: NativeTextInput | undefined | null;
 
   componentDidUpdate(prevProps: TextInputProps, prevState: State) {
+    const isFocusChanged = prevState.focused !== this.state.focused;
+    const isValueChanged = prevState.value !== this.state.value;
+    const isLabelLayoutChanged =
+      prevState.labelLayout !== this.state.labelLayout;
+    const isLabelChanged = prevProps.label !== this.props.label;
+    const isErrorChanged = prevProps.error !== this.props.error;
+
```
Comment (matkoson) [inline src/components/TextInput/TextInput.tsx:230]: Extracting logic into variables makes it easier for me to reason about a component.

**Comment by ravi-poonia** [comment]
Is there anything i can do to get it merged in master?

**Comment by matkoson** [comment]
@ravi-poonia Not at the moment. This PR is in queue for reviewing, we'll se after that. In the meantime you can review it yourself, if you wish.

File: src/components/IconButton.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ import { withTheme } from '../core/theming';
 
 import { Theme, $RemoveChildren } from '../types';
 
-type Props = $RemoveChildren<typeof TouchableRipple> & {
+export type IconButtonProps = $RemoveChildren<typeof TouchableRipple> & {
```
Comment (Trancever) [inline src/components/IconButton.tsx:18]: We don't need to export this type. If we need to use it in different place we should get it using `React.ComponentProps<typeof IconButton>`

Comment (jbinda) [reply src/components/IconButton.tsx]: I have update it as you said

**Comment by jbinda** [comment]
LGTM ! approved and merged to @Trancever branch and will test one more time before ship to master.

@matkoson great job, thanks !

**Pull request merged**